app/models/chef.rb

```diff
@@ -1,4 +1,11 @@
 class Chef < ApplicationRecord
    validates_presence_of :name
+   
    has_many :dishes
+
+   def all_ingredients
+    self.dishes.joins(:ingredients)
```

Comment:
If you add this additional relationship: `has_many :ingredients, through: :dishes` then you can just do `self.ingredients.distinct`